### PR TITLE
Many changes, fixes and additions involving new pattern scanning systems.

### PR DESCRIPTION
**Pull request by thisis2838**
Additions:
* `y_spt_set_collision_group` 
Sets collision group for player, useful for toggling Quickclip on and off
* `_y_spt_afterticks` 
Tick-based alternative to `_y_spt_afterframes`, useful for spamming keys
* `y_spt_hud_havok_velocity` 
Shows velocity and position of Havok hitbox
* `y_spt_hud_drawing_function`
Switches hud-drawing function pointers between SDK-derived and scanned ones, useful for if derived functions are wrong
* `y_spt_free_oob_movement` 
Enable/Disable free OOB movement
* `y_spt_disable_ammo_pickup_sound` 
Disables ammo pickup sounds, useful for when rendering demos
* `y_spt_disable_weapon_pickup_sound` 
Disables weapon pickup sounds, useful for when rendering demos
* `y_spt_hdtf_uncap` 
HDTF ONLY: disables velocity cap
* `y_spt_hdtf_viewroll` 
HDTF ONLY: disables view rolling

Patterns:
* New non-static pattern searching systems for finding entity offsets and backups for existing static signatures.

Fixes:
* `y_spt_additional_jumpboost` 
Broken due to function call order
* `PlayerRunCommand` offset determining
Previously erroneously using `FinishGravity`'s pattern index instead of `PlayerRunCommand`'s. [(Culprit commit)](https://github.com/YaLTeR/SourcePauseTool/blame/71292558e6cc6bff53071cd4e2f88fd56f246400/spt/OrangeBox/modules/ServerDLL.cpp#L451)
* Hud text drawing while no `screen` is available.
Defaulted text drawing position at 300 X.

### CODE REVIEW

**Comment by thekovic** [comment]
Lord be praised

**Comment by lipsanen** [comment]
been looking at the datatable stuff and been trying to implement the datatable stuff via GetDataDescMap() in the CBaseEntity vtable (so the extra scanning steps arent necessary) https://github.com/lipsanen/SourcePauseTool/tree/datatablestuff

semi scuffed implementation atm

**Comment by lipsanen** [comment]
Not too big of a fan that the added pattern scanning systems run in a single thread (current pattern searching solution is multithreaded) so that could cause some slowdown while loading the plugin.

"Disable ammo and weapon pickup sounds" - what's the problem with replacing the sound file with an empty sound?

I can try to implement the other cvars/fixes soonish assuming my mega refactor SPT goes through

**Comment by lipsanen** [comment]
PS. The CreateMove pattern for BMS is a substring of a previous pattern, what's up with that?

File: spt/OrangeBox/patterns.hpp [hunk 1]
```diff
@@ -198,7 +200,9 @@ namespace patterns
 		    "2257546",
 		    "55 8B EC 83 EC 54 53 56 8B 75 08 B8 ?? ?? ?? ?? F7 EE 57 03 D6 8B F9 C1 FA 06 8B CE 8B C2 C1 E8 1F",
 		    "2257546-hl1",
-		    "55 8B EC 83 EC 50 53 8B 5D 08 B8 ?? ?? ?? ?? F7 EB 56 03 D3 C1 FA 06 8B C2 C1 E8 1F 03 C2 8B D3");
+		    "55 8B EC 83 EC 50 53 8B 5D 08 B8 ?? ?? ?? ?? F7 EB 56 03 D3 C1 FA 06 8B C2 C1 E8 1F 03 C2 8B D3",
+			"BMS-Retail",
+			"55 8B EC 83 EC 54 53 56 8B 75 08 B8 ?? ?? ?? ?? F7 EE 57 03 D6 8B F9 C1 FA 06 8B CE 8B C2 C1 E8 1F 03 C2 6B C0 5A");
```
Comment (lipsanen) [inline spt/OrangeBox/patterns.hpp]: Why is this a substring of the 2257546 (whatever game/build on god's green earth is this) pattern?

Comment (thisis2838) [reply spt/OrangeBox/patterns.hpp]: I don't know, this might be Smiley's addition?

**Comment by thisis2838** [comment]
> "Disable ammo and weapon pickup sounds" - what's the problem with replacing the sound file with an empty sound?
Because it's unreliable and also some games store all resources in vpks

**Comment by thisis2838** [comment]
Closing this due to recent overhauls

**Pull request closed without merging**